java/vector/src/main/java/org/apache/arrow/vector/compression/CompressionUtility.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.vector.compression;
+
+import org.apache.arrow.flatbuf.BodyCompressionMethod;
+import org.apache.arrow.flatbuf.CompressionType;
+import org.apache.arrow.vector.ipc.message.ArrowBodyCompression;
+
+/**
+ * Utilities for data compression/decompression.
+ */
+public class CompressionUtility {
+
+  private CompressionUtility() {
+  }
+
+  /**
+   * Creates the {@link ArrowBodyCompression} object, given the {@link CompressionCodec}.
+   */
+  public static ArrowBodyCompression createBodyCompression(CompressionCodec codec) {
+    if (codec == null) {
+      return null;
+    }
+    for (int i = 0; i < CompressionType.names.length; i++) {
+      if (CompressionType.names[i].equals(codec.getCodecName())) {
```

Comment:
I feel like a switch statement and/or enum (similar to Type/ArrowType conversions would be better here than a tight loop of string comparisons. Not a huge deal as its a small array but I think its a bit easier to read.

Reply:
Thanks for your comment. I agree with you. 
The problem is that class `org.apache.arrow.flatbuf.CompressionType` is automatically generated by flatbuf, and it is not implemented by an enum. Instead, it has separate fields for ordinals and names. In addition, it only provides function to convert ordinals to names, but not vice versa.

Reply:
Agreed, I thought a pattern similar to `Types.java` (and an added enum) or a switch statement on the byte value might be more clear. Not a blocker for me though

Reply:
Thanks for the suggestion. I have revised the code to implement it through a switch statement. Please check if it looks good.

It looks clearer. However, we may need to change the code whenever we support a new compression method.